File: 0001-two-sum/0001-two-sum.cpp
// m3hu1
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    map<int,int> mpp;
	vector<int> ans;
    int n=nums.size();
	for (int i=0;i<n;i++)
	{
		int a=nums[i];
		int more= target-a;
		if (mpp.find(more) != mpp.end())
		{			
			ans.push_back( mpp[more]);
			ans.push_back(i);
			break;
		}
		mpp[a]=i;
	}
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = defaultdict(int)
        for idx, num in enumerate(nums):
            req = target - num
            if req in mp:
                return [idx, mp[req]]
            mp[num] = idx
        return [-1, -1]